evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -106,7 +106,8 @@ def test_make_pipeline_all_nan_no_categoricals(input_type, problem_type):
 
 @pytest.mark.parametrize("input_type", ["pd", "ww"])
 @pytest.mark.parametrize("problem_type", ProblemTypes.all_problem_types)
-def test_make_pipeline(input_type, problem_type):
+@pytest.mark.parametrize("prepended_components", [None, [], [DropColumns], [DropColumns, DropColumns]])
```

Comment:
Our linear component graph structure can't handle duplicate components with the same name. This means we can't have two "drop columns" actions in AutoMLSearch: 

```python
from evalml.data_checks import DataCheckAction, DataCheckActionCode
from evalml.demos import load_breast_cancer
from evalml.pipelines import BinaryClassificationPipeline
from evalml.automl import AutoMLSearch

X, y = load_breast_cancer()

class PipeLine(BinaryClassificationPipeline):
    component_graph = ["Random Forest Classifier"]
    

actions = [DataCheckAction(DataCheckActionCode.DROP_COL, details={"columns": ["mean radius"]}),
           DataCheckAction(DataCheckActionCode.DROP_COL, details={"columns": ["mean texture"]})]
automl = AutoMLSearch(X, y, problem_type="binary", data_check_actions=actions)
automl.search()
```

```python
Using default limit of max_batches=1.

Generating pipelines to search over...
*****************************
* Beginning pipeline search *
*****************************

Optimizing for Log Loss Binary. 
Lower score is better.

Using SequentialEngine to train and score pipelines.
Searching up to 1 batches for a total of 9 pipelines. 
Allowed model families: random_forest, extra_trees, decision_tree, catboost, lightgbm, linear_model, xgboost

Batch 1: (1/9) Mode Baseline Binary Classification P... Elapsed:00:00
	Starting cross validation
	Finished cross validation - mean Log Loss Binary: 12.868
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/sources/evalml/evalml/pipelines/component_graph.py in instantiate(self, parameters)
     77             try:
---> 78                 new_component = component_class(**component_parameters, random_seed=self.random_seed)
     79             except (ValueError, TypeError) as e:

~/sources/evalml/evalml/pipelines/components/transformers/column_selectors.py in __init__(self, columns, random_seed, **kwargs)
     15         if columns and not isinstance(columns, list):
---> 16             raise ValueError(f"Parameter columns must be a list. Received {type(columns)}.")
     17 

ValueError: Parameter columns must be a list. Received <class 'str'>.

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
<ipython-input-50-39598b64bdb5> in <module>
     13            DataCheckAction(DataCheckActionCode.DROP_COL, details={"columns": ["mean texture"]})]
     14 automl = AutoMLSearch(X, y, problem_type="binary", data_check_actions=actions)
---> 15 automl.search()

~/sources/evalml/evalml/automl/automl_search.py in search(self, show_iteration_plot)
    508             try:
    509                 if not loop_interrupted:
--> 510                     current_batch_pipelines = self._automl_algorithm.next_batch()
    511             except StopIteration:
    512                 logger.info('AutoML Algorithm out of recommendations, ending')

~/sources/evalml/evalml/automl/automl_algorithm/iterative_algorithm.py in next_batch(self)
     63         next_batch = []
     64         if self._batch_number == 0:
---> 65             next_batch = [pipeline_class(parameters=self._transform_parameters(pipeline_class, {}), random_seed=self.random_seed)
     66                           for pipeline_class in self.allowed_pipelines]
     67 

~/sources/evalml/evalml/automl/automl_algorithm/iterative_algorithm.py in <listcomp>(.0)
     63         next_batch = []
     64         if self._batch_number == 0:
---> 65             next_batch = [pipeline_class(parameters=self._transform_parameters(pipeline_class, {}), random_seed=self.random_seed)
     66                           for pipeline_class in self.allowed_pipelines]
     67 

~/sources/evalml/evalml/pipelines/classification_pipeline.py in __init__(self, parameters, random_seed)
     23         """
     24         self._encoder = LabelEncoder()
---> 25         super().__init__(parameters, random_seed=random_seed)
     26 
     27     def fit(self, X, y):

~/sources/evalml/evalml/pipelines/pipeline_base.py in __init__(self, parameters, random_seed)
     77         else:
     78             self._component_graph = ComponentGraph(component_dict=self.component_graph, random_seed=self.random_seed)
---> 79         self._component_graph.instantiate(parameters)
     80 
     81         self.input_feature_names = {}

~/sources/evalml/evalml/pipelines/component_graph.py in instantiate(self, parameters)
     80                 self._is_instantiated = False
     81                 err = "Error received when instantiating component {} with the following arguments {}".format(component_name, component_parameters)
---> 82                 raise ValueError(err) from e
     83 
     84             component_instances[component_name] = new_component

ValueError: Error received when instantiating component Drop Columns Transformer with the following arguments {'columns': 'mean texture'}
```

I think this should be resolved before merge. Our data checks are currently expected to output multiple actions (id, highly null, target leakage) but if AutoML can't support that after this PR then adding the actions param to AutoMLSearch seems premature imo.

Let me know what you think!

Reply:
Thank you for pointing this out! I was scratching my head as to why this test seemed to pass, but it's `search` that breaks. If we don't choose to combine components, or even if we do, a possible solution is to add custom names like 'Drop Columns Selector from DataCheckAction' or something 🤔

Reply:
We should probably file an issue for this too hehe I think we should handle the duplicate component name case a bit better hehe

Reply:
Just filed https://github.com/alteryx/evalml/issues/2015 to track the IterativeAlgorithm bug. I think that issue is blocking this PR but we can talk about it in stand-up hehe

Reply:
This thread totally rocks. Great hunting and thanks for getting that issue fixed @angela97lin !